apps/proto/lib/lexical/proto/macros/struct.ex

```diff
@@ -23,7 +25,10 @@ defmodule Lexical.Proto.Macros.Struct do
     quote location: :keep do
       @enforce_keys unquote(required_keys)
       defstruct unquote(keys)
+      unquote(Typespec.keyword_constructor_options(opts, env, :option, :options))
```

Comment:
This would get hoisted above the `quote`, so it would be:

```elixir
{option_typespec, options_typespec} = Typespec.keyword_constructor_options(opts, env)

quote location: :keep do
  @enforce_keys unquote(required_keys)
  defstruct unquote(keys)
  @type option :: unquote(option_typespec)
  @type options :: unquote(options_typespec)
```